client/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx

```diff
@@ -1,22 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Button from '@material-ui/core/Button';
 import QuestionsDummyData from '../DummyData/QuestionsDummyData';
 import QAndA from './QAndAComponents/QAndA.jsx';
 import QuestionSearch from './QAndAComponents/QuestionSearch.jsx';
 import Photos from './QAndAComponents/Photos.jsx';
+import config from '../../../config';
+import axios from 'axios';
 
-const QuestionsAndAnswers = () => {
-  var sortedQuestions = sortByHelpfulness(QuestionsDummyData.questions.results, 4);
+const getQuestions = (id) => {
+  return axios
+    .get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/qa/questions?product_id=${id}`, {
+      headers: {
+        Authorization: config.GITHUB_TOKEN,
+      },
+    })
+    .then((response) => {
+      return response.data;
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+const QuestionsAndAnswers = ( { productId }) => {
+  const [questions, setQuestions] = useState([]);
+  const [shownQuestions, setShownQuestions] = useState([]);
+
+  useEffect(() => {
+    async function fetchQuestions() {
+      const questionData = await getQuestions(productId);
+      setQuestions(questionData.results);
+      setShownQuestions(questionData.results);
+    }
+    fetchQuestions();
+  }, []);
+
+  const questionSearch = (searchInput) => {
+    if (searchInput.length < 3) {
+      setShownQuestions(questions);
+      return;
+    }
+    var searchMatchingQuestions = [];
+    for (var i = 0; i < questions.length; i++) {
+      if (questions[i].question_body.includes(searchInput)) {
+        searchMatchingQuestions.push(questions[i]);
+      }
+    }
+    setShownQuestions(searchMatchingQuestions);
+  };
+
+  var sortedQuestions = sortQuestionsByHelpfulness(shownQuestions, 4);
```

Comment:
Should the default number of questions per the spec be 2 rather than 4: "The list will by default only display up to 2 questions asked.   If there are more than 2 questions that have been asked for the given product, a button for “More Answered Questions” will appear below the list."

Reply:
Good catch!  I will change it to 2 on the next PR.  I noticed the following quote in the requirements document, which is why I chose 4, but I like the look of 2 better (plus it matches the adobe mockup).  "By default, on page load up to four questions should be displayed. Up to two answers should display for each question."